### ISSUE TITLE
Creating multiple font sizes, best / recommended practice?

### ISSUE THREAD

**Issue by inflex** [ocornut/imgui#5616]
I've been Valgrinding my code a bit and I've notice it seems that my font atlas is consuming about 130MB of memory, though I'm not sure if that's specifically good/bad.

Essentially I have 10 font sizes, each 25% larger than the previous, in regular and bold variants ( so 20 altogether ).

Here's a typical entry in the loop;
```
        // app.fb[] - bold fonts
        // app.f[] - regular fonts
        //
	app.fb[i] = app.io.Fonts->AddFontFromMemoryCompressedTTF((void *)Roboto_Black_compressed_data, Roboto_Black_compressed_size, ft_size, NULL, ranges);
```

Is there a superior way I should be doing this?

Apologies if this question isn't defined sufficiently specifically.

**Comment by ocornut** [comment]
You can reduce the OversampleH values from 3 to 1 for larger fonts. It’s only useful for very small / thin fonts.

**Comment by inflex** [comment]
As an update; moving to a lower OversampleH only reduced the consumption slightly, but the real culprit was my merged MaterialIcons range;  many moons ago I had simply included everything from the minimum to maximum address of icon I required, which was about 0x0800 icons.

Taking 5 minutes to specifically select only the ones I needed dropped my memory consumption from 130MB to 30MB :| 

```
	static const ImWchar icon_ranges[] = 
	{
		0xe148, 0xe148,
		0xe14c, 0xe14c,
		0xe15d, 0xe15d,
		0xe3bc, 0xe3bc, 
		0xe3dc, 0xe3dc,
		0xe419, 0xe419,
		0xe41a, 0xe41a,
		0xe875, 0xe875, 
		0xe8d5, 0xe8d5,
		0xe8ff, 0xe8ff,
		0xe900, 0xe900,
		0 
};

	app.f[i] = app.io.Fonts->AddFontFromMemoryTTF((void *)MaterialIcons_data, MaterialIcons_size, ft_size,  &icon_config, icon_ranges);
```

Thank you for the tips in the previous reply though, as they also helped me solve a few other small issues I was encountering with my application's appearance.

**Comment by ocornut** [comment]
As a suggestion, it may be syntaxically easier to use a ImFontGlyphRangesBuilder to create that range.

**Comment by inflex** [comment]
I presume you're referring to the AddChar() from this document - 

https://github.com/ocornut/imgui/blob/master/docs/FONTS.md#using-custom-glyph-ranges

Thanks, I might rewrite the code to make it more durable / easier-to-read.  I was manually transferring the character index from the font header file.

**Comment by ocornut** [comment]
Dynamic font support was now merged in #8465. Custom glyph ranges should be unnecessary.
Will amends docs/faq/comments with the probable wave of feedback.
